key-manager-test/src/test/scala/org/bitcoins/keymanager/config/KeyManagerAppConfigTest.scala

```diff
@@ -0,0 +1,110 @@
+    val overrider = ConfigFactory.parseString(s"""
+                                                 |bitcoin-s {
+                                                 |  network = mainnet
+                                                 |}
+                                                 |""".stripMargin)
+
+    val throughConstructor = KeyManagerAppConfig(tempDir, overrider)
+    val throughWithOverrides = config.withOverrides(overrider)
+    assert(throughWithOverrides.network == MainNet)
+    assert(throughWithOverrides.network == throughConstructor.network)
+
+    assert(throughWithOverrides.datadir == throughConstructor.datadir)
+
+  }
+
+  it must "be overridable without screwing up other options" in {
+    val otherConf = ConfigFactory.parseString(
+      s"bitcoin-s.keymanager.bip39password = passA"
+    )
+    val thirdConf =
+      ConfigFactory.parseString(s"bitcoin-s.keymanager.bip39password = passB")
+
+    val overridden = config.withOverrides(otherConf)
+
+    val twiceOverridden = overridden.withOverrides(thirdConf)
+
+    assert(overridden.bip39PasswordOpt.contains("passA"))
+    assert(twiceOverridden.bip39PasswordOpt.contains("passB"))
+
+    assert(config.datadir == overridden.datadir)
+    assert(twiceOverridden.datadir == overridden.datadir)
+  }
+
+  it must "be overridable with multiple levels" in {
+    val testnet = ConfigFactory.parseString("bitcoin-s.network = testnet3")
+    val mainnet = ConfigFactory.parseString("bitcoin-s.network = mainnet")
+    val overridden: KeyManagerAppConfig = config.withOverrides(testnet, mainnet)
+    assert(overridden.network == MainNet)
+  }
+
+  it must "have user data directory configuration take precedence" in {
+    val tempFile = Files.createFile(tempDir.resolve("bitcoin-s.conf"))
+    val confStr = """
+                    | bitcoin-s {
+                    |   network = testnet3
+                    |
+                    |   logging {
+                    |     level = off
+                    |
+                    |     p2p = warn
+                    |   }
+                    | }
+    """.stripMargin
+    val _ = Files.write(tempFile, confStr.getBytes())
+
+    val appConfig = KeyManagerAppConfig(directory = tempDir)
+
+    assert(appConfig.datadir == tempDir.resolve("testnet3"))
+    assert(appConfig.network == TestNet3)
+  }
+
+  it must "move an old seed into the seeds folder" in {
+    // generate and write mnemonic
+    val mnemonicCode = CryptoGenerators.mnemonicCode.sampleSome
+    val mnemonic = DecryptedMnemonic(mnemonicCode, TimeUtil.now)
+    val seedPath = tempDir.resolve(WalletStorage.ENCRYPTED_SEED_FILE_NAME)
+    WalletStorage.writeMnemonicToDisk(seedPath, mnemonic)
+
+    config.start().map { _ =>
+      assert(Files.exists(seedPath), "We should not delete the old seed!")
+      assert(
+        Files.exists(
+          tempDir
+            .resolve(WalletStorage.SEED_FOLDER_NAME)
+            .resolve(WalletStorage.ENCRYPTED_SEED_FILE_NAME)))
+    }
+  }
```

Comment:
probably should add a test case just making sure we can have multiple seeds in the `seeds` directory and they get written to the write place?